### PR TITLE
HDDS-7145. Recon:Show last OM DB sync time on Overview page

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
To Display OM Last Updated Timestamp and Snapshot changes

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7145

## How was this patch tested?
Manually Tested

### CODE REVIEW

**Comment by kerneltime** [comment]
cc @dombizita

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx [hunk 1]
```diff
@@ -38,21 +42,39 @@ class AutoReloadPanel extends React.Component<IAutoReloadPanelProps> {
...
+                 {lastUpdateOMSyncText} : {moment(lastUpdateOMSync).format('ll LTS')}</span>
+    const lastUpdatedText = lastUpdated === 0 || lastUpdated === undefined? 'NA' :
       (
         <Tooltip
           placement='bottom' title={moment(lastUpdated).format('ll LTS')}
         >
           {moment(lastUpdated).format('LTS')}
         </Tooltip>
       );
+      const lastUpdatedOMInfo = lastUpdatedOM === 0 || lastUpdatedOM === undefined || lastUpdateOMSync===0 || lastUpdateOMSync=== undefined ? 'NA' :
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:56]: nit
```suggestion
      const lastUpdatedOMInfo = lastUpdatedOM === 0 || lastUpdatedOM === undefined || lastUpdateOMSync === 0 || lastUpdateOMSync === undefined ? 'NA' :
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 2]
```diff
@@ -86,11 +95,15 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
...
+      axios.get('/api/v1/containers/missing'),
+      axios.get('/api/v1/task/status')
+    ]).then(axios.spread((clusterStateResponse, missingContainersResponse,taskstatusResponse) => {
+      
       const clusterState: IClusterStateResponse = clusterStateResponse.data;
-      const missingContainers: IMissingContainersResponse = missingContainersResponse.data;
+      const missingContainers: IMissingContainersResponse = missingContainersResponse.data;     
+      const taskStatus = taskstatusResponse.data && taskstatusResponse.data.filter((item:any) => item.taskName === 'OmDeltaRequest' || item.taskName === 'OmSnapshotRequest').sort((c1:any, c2:any) => c2.lastUpdatedTimestamp - c1.lastUpdatedTimestamp);
       const missingContainersCount = missingContainers.totalCount;
+      console.log("radha2",taskStatus);
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:106]: nit: debugging message
```suggestion
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx [hunk 3]
```diff
@@ -38,21 +42,39 @@ class AutoReloadPanel extends React.Component<IAutoReloadPanelProps> {
...
+      (
+        <Tooltip
+          placement='bottom' title={textOMSync}
+        >
+          {moment(lastUpdatedOM).format('LTS')}
+        </Tooltip>
+      );
+     const lastUpdatedOMDisplay= lastUpdatedOM === 0 || lastUpdatedOM === undefined || lastUpdateOMSync===0 || lastUpdateOMSync=== undefined ? '' :
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:64]: nit
```suggestion
     const lastUpdatedOMDisplay= lastUpdatedOM === 0 || lastUpdatedOM === undefined || lastUpdateOMSync === 0 || lastUpdateOMSync === undefined ? '' :
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 4]
```diff
@@ -75,7 +79,12 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
       buckets: 0,
       keys: 0,
       missingContainersCount: 0,
-      lastUpdated: 0
+      lastUpdated: 0,
+      lastUpdatedOM:0,
+      lastUpdateOMSync:0,
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:84]: Rename this to `lastUpdatedOMDBDelta` and `lastUpdatedOMDBFull` respectively.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 5]
```diff
@@ -86,11 +95,15 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
...
-      axios.get('/api/v1/containers/missing')
-    ]).then(axios.spread((clusterStateResponse, missingContainersResponse) => {
+      axios.get('/api/v1/containers/missing'),
+      axios.get('/api/v1/task/status')
+    ]).then(axios.spread((clusterStateResponse, missingContainersResponse,taskstatusResponse) => {
+      
       const clusterState: IClusterStateResponse = clusterStateResponse.data;
-      const missingContainers: IMissingContainersResponse = missingContainersResponse.data;
+      const missingContainers: IMissingContainersResponse = missingContainersResponse.data;     
+      const taskStatus = taskstatusResponse.data && taskstatusResponse.data.filter((item:any) => item.taskName === 'OmDeltaRequest' || item.taskName === 'OmSnapshotRequest').sort((c1:any, c2:any) => c2.lastUpdatedTimestamp - c1.lastUpdatedTimestamp);
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:104]: Can we use two variables to hold values from `OmDeltaRequest` and `OmSnapshotRequest` separately so the logic can be cleaner? Thx.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 6]
```diff
@@ -101,7 +114,12 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
         buckets: clusterState.buckets,
         keys: clusterState.keys,
         missingContainersCount,
-        lastUpdated: Number(moment())
+        lastUpdated: Number(moment()),
+        lastUpdatedOM: taskStatus[0] ? taskStatus[0] && taskStatus[0].lastUpdatedTimestamp : 0,
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:118]: Can we simply use `taskStatus[0].lastUpdatedTimestamp`?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 7]
```diff
@@ -101,7 +114,12 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
...
+        lastUpdateOMSync:taskStatus[1] ? taskStatus[1] && taskStatus[1].lastUpdatedTimestamp : 0,
+        lastUpdatedOMText:taskStatus[0] ? taskStatus[0] && taskStatus[0].taskName.toLowerCase() === 'OmDeltaRequest'.toLowerCase() ? 'Last Delta Update': 'Last Full Update': '',
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:120]: I believe we could simply use `Last Delta Update` once we separate the two variables?

**Review by smengcl — CHANGES_REQUESTED** [review]
Thanks @smitajoshi12 for the patch.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx (hunk 1 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:56]: Will do

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx (hunk 2 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:106]: Will do

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx (hunk 3 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/autoReloadPanel/autoReloadPanel.tsx:64]: Will do respective changes

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx (hunk 4 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:84]: Will do Respective changes

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx (hunk 5 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:104]: Will do respective changes

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx (hunk 7 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx]: Done Changes

**Pull request closed without merging**